contracts/contracts/OpenSignal.sol

```diff
@@ -48,12 +49,13 @@ contract OpenSignal is ERC2771Context, ReentrancyGuard {
         _;
     }
 
-    constructor(address _trustedForwarder, IERC20 _nativeToken) ERC2771Context(_trustedForwarder) {
+    constructor(address _trustedForwarder, IERC20 _nativeToken, address _rewardsDistribution ) ERC2771Context(_trustedForwarder) {
```

Comment:
Where is `_rewardsDistribution` used?

Reply:
It's used to get instance of rewardDistribution which is then called everytime a token is staked/unstaked.

Reply:
You're making the instance on line 58, that is what is used in the contract, both approaches are fine but currently this variable is not in use

Reply:
Ah yep, this is now resolved. I changed it so that we pass in the address of the already deployed contract.